neard/src/runtime.rs

```diff
@@ -217,47 +216,49 @@ impl NightshadeRuntime {
         state_file.push(STATE_DUMP_FILE);
-        self.store
-            .load_from_file(ColState, state_file.as_path())
-            .expect("Failed to read state dump");
-        let mut roots_files = self.home_dir.clone();
+        store.load_from_file(ColState, state_file.as_path()).expect("Failed to read state dump");
+        let mut roots_files = home_dir.to_path_buf();
         roots_files.push(GENESIS_ROOTS_FILE);
         let mut file = File::open(roots_files).expect("Failed to open genesis roots file.");
         let mut data = vec![];
         file.read_to_end(&mut data).expect("Failed to read genesis roots file.");
         let state_roots: Vec<StateRoot> =
             BorshDeserialize::try_from_slice(&data).expect("Failed to deserialize genesis roots");
-        (self.store.clone(), store_update, state_roots)
+        (store, state_roots)
     }
 
-    fn genesis_state_from_records(&self) -> (Arc<Store>, StoreUpdate, Vec<StateRoot>) {
-        let mut store_update = self.store.store_update();
+    fn genesis_state_from_records(
+        store: Arc<Store>,
+        genesis: Arc<Genesis>,
+    ) -> (Arc<Store>, ShardTries, Vec<StateRoot>) {
+        let mut store_update = store.store_update();
         let mut state_roots = vec![];
-        let num_shards = self.genesis.config.num_block_producer_seats_per_shard.len() as NumShards;
+        let num_shards = genesis.config.num_block_producer_seats_per_shard.len() as NumShards;
         let mut shard_records: Vec<Vec<StateRecord>> = (0..num_shards).map(|_| vec![]).collect();
         let mut has_protocol_account = false;
-        for record in self.genesis.records.as_ref() {
+        for record in genesis.records.as_ref() {
             shard_records[state_record_to_shard_id(record, num_shards) as usize]
                 .push(record.clone());
             if let StateRecord::Account { account_id, .. } = record {
-                if account_id == &self.genesis.config.protocol_treasury_account {
+                if account_id == &genesis.config.protocol_treasury_account {
                     has_protocol_account = true;
                 }
             }
         }
         assert!(has_protocol_account, "Genesis spec doesn't have protocol treasury account");
+        let num_shards = genesis.config.num_block_producer_seats_per_shard.len() as NumShards;
+        let tries = ShardTries::new(store.clone(), num_shards);
+        let runtime = Runtime::new(genesis.config.runtime_config.clone());
         for shard_id in 0..num_shards {
-            let validators = self
-                .genesis
+            let validators = genesis
                 .config
                 .validators
                 .iter()
                 .filter_map(|account_info| {
-                    if self.account_id_to_shard_id(&account_info.account_id) == shard_id {
+                    if account_id_to_shard_id(&account_info.account_id, num_shards) == shard_id {
```

Comment:
`account_id_to_shard_id` is about to start depending on an epoch id, and this calling to `account_id_to_shard_id` outside of runtime adapter is deprecated.

Reply:
I don't think it will affect his change. The number of shards in genesis is predetermined anyways

Reply:
We can store and extract mapping account_id -> shard_id separately.

However, there is no reason to call deprecated function because I'll revert it later anyway. We can also add a comment for `account_id_to_shard_id` function like "// TODO #2961: Deprecated, use runtime adapter instead".

Reply:
@Kouprin Bowen calls it here because he doesn't have the runtime adapter here anymore.
But he needs specifically the `account_id_to_shard_id` as of genesis, so he doesn't need the part of the logic that determines the epoch.

Reply:
@SkidanovAlex we can put it into Store then.

Reply:
@Kouprin I don't know how exactly you plan to change it, but whatever you do, shouldn't there be a function like `genesis_account_id_to_shard_id`? It seems that you always have to treat genesis specially.

Reply:
I guess we can avoid using several methods doing the same. The plan is moving `account_id_to_shard_id` into RuntimeAdapter, then into EpochManager, and then into Store. So, in Store we can have methods `account_id_to_shard_id` and `num_shards` which called with known Epoch id. In case of Epoch id = default, we read genesis data from Store.

Reply:
I see. We have to Store here, right? It seems we need to keep mapping account_id to shard_id in config.